controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -941,6 +1121,44 @@ func (pp *portPublisher) unsubscribe(listener EndpointUpdateListener) {
 	pp.metrics.setSubscribers(len(pp.listeners))
 }
 
+// getServerOpaquePodPorts gets all pods that are selected by Servers with an
+// opaque protocol. This is called when when new Endpoints or EndpointSlices
+// are added and is used to check if any of the new endpoints are selected by
+// a Server that would change the protocol that the backing pod expects.
+func (pp *portPublisher) getServerOpaquePodPorts() map[podPortID]struct{} {
+	podPorts := make(map[podPortID]struct{})
+	objects := pp.k8sAPI.Srv().Informer().GetIndexer().List()
```

Comment:
Can we use the Lister instead of the Informer?

Reply:
Yep meant to use a Lister here